src/frontend/components/modal.jsx

```diff
@@ -0,0 +1,88 @@
+import { useRef, useEffect } from "react";
+import { RiUploadCloud2Fill } from "react-icons/ri";
+import { GiCancel } from "react-icons/gi";
+import {useUser} from "frontend/context/userContext.jsx";
+
+const Modal = ({ showModal, setShowModal, editUserData, setEditUserData }) => {
+    const imageRef = useRef();
+    const {loggedUserData} = useUser();
+
+    const uploadImage = () => {
+        imageRef.current.click();
+    }
+
+    const updateUserData = () => {
+        setShowModal(false)
+    }
+
+    useEffect(() => {
+        localStorage.setItem("edited-data" ,JSON.stringify(editUserData))
+    })
+
+    return (
+        <>
+            {showModal ? (
```

Comment:
can simplify with && instead of a ternary.

Reply:
Done